Sources/SwiftKafka/Configuration/KafkaTransactionalProducerConfiguration.swift

```diff
@@ -0,0 +1,134 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the swift-kafka-gsoc open source project
+//
+// Copyright (c) 2022 Apple Inc. and the swift-kafka-gsoc project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of swift-kafka-gsoc project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+// FIXME: should we really duplicate `KafkaProducerConfiguration`
+// FIXME: after public api updated?
+public struct KafkaTransactionalProducerConfiguration {
+    // MARK: - SwiftKafka-specific Config properties
+
+    /// The time between two consecutive polls.
+    /// Effectively controls the rate at which incoming events are consumed.
+    /// Default: `.milliseconds(100)`
+    public var pollInterval: Duration = .milliseconds(100)
+
+    /// Maximum timeout for flushing outstanding produce requests when the ``KakfaProducer`` is shutting down.
+    /// Default: `10000`
+    public var flushTimeoutMilliseconds: Int = 10000 {
+        didSet {
+            precondition(
+                0...Int(Int32.max) ~= self.flushTimeoutMilliseconds,
+                "Flush timeout outside of valid range \(0...Int32.max)"
+            )
+        }
+    }
+
+    // MARK: - Producer-specific Config Properties
+
+    /// When set to true, the producer will ensure that messages are successfully produced exactly once and in the original produce order. The following configuration properties are adjusted automatically (if not modified by the user) when idempotence is enabled: max.in.flight.requests.per.connection=5 (must be less than or equal to 5), retries=INT32_MAX (must be greater than 0), acks=all, queuing.strategy=fifo. Producer instantation will fail if user-supplied configuration is incompatible.
+    /// Default: `false`
+    internal let enableIdempotence: Bool = true
```

Comment:
The difference between `KafkaProducerConfiguration` and `KafkaTransactionalProducerConfiguration` is in couple of properties:
1. `enableIdempotence` always `true` for transactions
2. `maxInFlightRequestsPerConnection` not greater than `5`
3. `transactionsTimeout` is set to socket timeout or greater than socket timeout